client/state/sites/updates/reducer.js

```diff
@@ -19,42 +23,97 @@ import {
 
 import { itemsSchema } from './schema';
 
-export const items = createReducer(
-	{},
-	{ [ SITE_UPDATES_RECEIVE ]: ( state, { siteId, updates } ) => Object.assign( {}, state, { [ siteId ]: updates } ) },
-	itemsSchema
-);
+export const items = ( state = {}, action ) => {
+	switch ( action.type ) {
+		case SITE_RECEIVE:
+		case SITES_RECEIVE:
+		case SITES_UPDATE:
+			// Normalize incoming site(s) to array
+			const sites = action.site ? [ action.site ] : action.sites;
+			forEach( sites, ( site ) => {
+				const updates = Object.assign(
+					{},
+					site.updates,
+					{
+						wp_version: get( site, 'options.software_version' ),
+						jp_version: get( site, 'options.jetpack_version' )
+					}
+				);
+
+				merge( state,
```

Comment:
I don't get it why we are merging `state` object twice in the same call. Is it needed?